### DISCUSSION TITLE
Supported formats/containers?

### DISCUSSION THREAD

**Discussion by DavidCWGA** [beetbox/beets#3964] [Q&A] [answered]
MKV containers are not supported and are ignored by the importer - where can I find a list of formats that *are* supported?

Specifically, I have some music that is encoded in DTS, and is currently inside an MKV container, but beets refuses to touch it. MKA is similarly ignored. I'm happy to put it inside a different container but I would rather avoid transcoding it.

I did look through the documentation but I apologize if I missed it.

**Comment by jackwilsdon** [comment, marked as answer]
From what I can tell, we pretty much support everything mutagen does (the library we use for parsing files). To be more specific, here's the list of formats our `mediafile` library can handle (from [here](https://github.com/beetbox/mediafile/blob/7ecd86101ec3344e7fb25bbf781cd7f86646aa9d/mediafile.py#L1520-L1545)):

* M4A
* MP4
* MP3
* FLAC
* OggOpus (opus)
* OggVorbis (ogg)
* MonkeysAudio (ape)
* WavPack (wv)
* MusePack (mpc)
* ASF
* AIFF
* DSF

I'm not sure if all of these have been tested with beets though. For reference, it seems like https://github.com/quodlibet/mutagen/issues/3 discusses implementing MKA support in mutagen (although it hasn't progressed in a while).

Also related: #3729.

**Comment by DavidCWGA** [comment]
OK thanks. It would be nice if this list was in the documentation.

The trick then is put DTS tracks into an MP4 container, which is supported by MP4, and although beets misidentifies them as "AAC" tracks, it doesn't mess them up in any way and they still play just fine. Although I did have to pre-tag the files with Picard before importing otherwise beets mis-matched them into an apparently random order that I don't really understand.

So maybe some feature requests?

1. Analyze the contents of MP4 files and don't assume they are AAC by default
2. Support MKV or at least MKA?

Although I understand that some of this relies on mutagen getting updated first.

Thanks.

**Reply by jackwilsdon** [reply]
>misidentifies them as "AAC" tracks

Where is it you're seeing this? As you suggested, it's posisble we're just assuming all MP4s are AAC (which is not true 100% of the time).

**Reply by DavidCWGA** [reply]
It identifies them as "AAC" during the import process (the album is a mix of FLAC and DTS-inside-MP4), and after import, "$format" on the track returns "AAC". The code linked above is pretty clear, it treats MP4 and M4A files the same - it looks inside to see if it contains ALAC, and if it doesn't, it assumes that it is AAC.

Admittedly, storing DTS inside an MP4 container is kind of a hack, but it is permitted inside an MP4 container, and as far as I can tell it's the only container that beets will import that can contain DTS.
